### PR TITLE
[DevEx] xpkg: add error contexts and print & fail on parse errors

### PR DESCRIPTION
**Pull request by sttts**
This PR adds errors contexts (e.g. filenames) to errors returned in the xpkg commands. Moreover, it removes the TODO in the workspace loader of missing error handling.

### CODE REVIEW

**Review by tnthornton — COMMENTED** [review]
Thanks @sttts ! I love all of the improvements to the amount of context we're providing. I left a couple of questions below. Let me know if there's anything I can clarify 👍

File: cmd/up/xpkg/build.go [hunk 1]
```diff
@@ -115,6 +115,35 @@ type buildCmd struct {
 	Ignore       []string `help:"Paths, specified relative to --package-root, to exclude from the package."`
 }
 
+func (c *buildCmd) Help() string {
```
[the remark this answers is not in the document]
Comment (sttts) [reply cmd/up/xpkg/build.go:118]: am actually calling this in the `xpkg` command. But let me reread and maybe rephrase one of them.

[the remark this answers is not in the document]
Comment (sttts) [reply cmd/up/xpkg/build.go:118]: done

File: internal/xpkg/workspace/workspace.go [hunk 2]
```diff
@@ -238,15 +255,24 @@ func (v *View) parseDoc(ctx context.Context, pCtx parseContext) (NodeIdentifier,
...
-	// NOTE(hasheddan): unmarshal returns an error if Kind is not defined.
 	// TODO(hasheddan): we cannot make use of strict unmarshal to identify
 	// extraneous fields because we don't have the underlying Go types. In
 	// the future, we would like to provide warnings on fields that are
 	// extraneous, but we will likely need to augment the OpenAPI validation
 	// to do so.
 	if err := k8syaml.Unmarshal(b, &obj); err != nil {
-		return NodeIdentifier{}, err
+		if v.printer != nil {
+			v.printer.Printfln("WARNING: ignoring file %s: missing 'kind' field, not a Kubernetes object", v.relativePath(pCtx.path))
```
[the remark this answers is not in the document]
Comment (sttts) [reply internal/xpkg/workspace/workspace.go:273]: Added error aggregation here too now.

File: cmd/up/xpkg/build.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sttts) [reply cmd/up/xpkg/build.go:118]: Please take a look at the new text for the build command. You will probably have more background of the command, so feel free to suggest another text, fix or add things.

File: internal/xpkg/workspace/workspace.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sttts) [reply internal/xpkg/workspace/workspace.go:273]: To understand: for the language server we want to be more permissive of errors. When the user edits files, we should try to keep going in passing. Thinking about adding a permissive flag that ignores errors, just warns.

[the remark this answers is not in the document]
Comment (sttts) [reply internal/xpkg/workspace/workspace.go]: done

**Review by tnthornton — APPROVED** [review]
LGTM 👍. Thanks again @sttts!

**Review by ulucinar — APPROVED** [review]
Thanks @sttts, just gave this PR a try for:
- `upbound/provider-aws`: `up xpkg batch` is producing provider families with the expected layers and metadata. Observed with: `index.docker.io/ulucinar/provider-family-aws:v0.37.0-rc.0.56.g257bedd2`, `index.docker.io/ulucinar/provider-aws-ec2:v0.37.0-rc.0.56.g867782ef`, `index.docker.io/ulucinar/provider-aws:v0.37.0-rc.0.56.g867782ef`
- `upbound/provider-azuread`: `up xpkg build` and `up xpkg push`  are producing provider packages with the expected layers and metadata. Observed with: `index.docker.io/ulucinar/provider-azuread:v0.10.0-rc.0.6.gb7fc503`

Thank you for the improvements.

**Comment by sttts** [comment]
Thank @ulucinar. Let's 🚢  it.
